datahub-web-react/src/app/ingest/source/builder/RecipeForm/common.tsx

```diff
@@ -0,0 +1,420 @@
+export const VIEW_DENY: RecipeField = {
+    name: 'view_pattern.deny',
+    label: 'Deny Patterns',
+    tooltip: 'Use regex here.',
+    type: FieldType.LIST,
+    buttonLabel: 'Add pattern',
+    fieldPath: viewDenyFieldPath,
+    rules: null,
+    section: 'Views',
+    setValueOnRecipeOverride: (recipe: any, values: string[]) =>
+        setListValuesOnRecipe(recipe, values, viewDenyFieldPath),
+};
+
+const tableAllowFieldPath = 'source.config.table_pattern.allow';
+export const TABLE_ALLOW: RecipeField = {
+    name: 'table_pattern.allow',
+    label: 'Allow Patterns',
+    tooltip: 'Use regex here.',
+    type: FieldType.LIST,
+    buttonLabel: 'Add pattern',
+    fieldPath: tableAllowFieldPath,
+    rules: null,
+    section: 'Tables',
+    setValueOnRecipeOverride: (recipe: any, values: string[]) =>
+        setListValuesOnRecipe(recipe, values, tableAllowFieldPath),
+};
+
+const tableDenyFieldPath = 'source.config.table_pattern.deny';
+export const TABLE_DENY: RecipeField = {
+    name: 'table_pattern.deny',
+    label: 'Deny Patterns',
+    tooltip: 'Use regex here.',
+    type: FieldType.LIST,
+    buttonLabel: 'Add pattern',
+    fieldPath: tableDenyFieldPath,
+    rules: null,
+    section: 'Tables',
+    setValueOnRecipeOverride: (recipe: any, values: string[]) =>
+        setListValuesOnRecipe(recipe, values, tableDenyFieldPath),
+};
+
+const chartAllowFieldPath = 'source.config.chart_pattern.allow';
+export const CHART_ALLOW: RecipeField = {
+    name: 'chart_pattern.allow',
+    label: 'Allow Patterns',
+    tooltip: 'Use regex here.',
+    type: FieldType.LIST,
+    buttonLabel: 'Add pattern',
+    fieldPath: chartAllowFieldPath,
+    rules: null,
+    section: 'Charts',
+    setValueOnRecipeOverride: (recipe: any, values: string[]) =>
+        setListValuesOnRecipe(recipe, values, chartAllowFieldPath),
+};
+
+const chartDenyFieldPath = 'source.config.chart_pattern.deny';
+export const CHART_DENY: RecipeField = {
+    name: 'chart_pattern.deny',
+    label: 'Deny Patterns',
+    tooltip: 'Use regex here.',
+    type: FieldType.LIST,
+    buttonLabel: 'Add pattern',
+    fieldPath: chartDenyFieldPath,
+    rules: null,
+    section: 'Charts',
+    setValueOnRecipeOverride: (recipe: any, values: string[]) =>
+        setListValuesOnRecipe(recipe, values, chartDenyFieldPath),
+};
+
+const topicAllowFieldPath = 'source.config.topic_patterns.allow';
+export const TOPIC_ALLOW: RecipeField = {
+    name: 'topic_patterns.allow',
+    label: 'Allow Patterns',
+    tooltip: 'Use regex here.',
+    type: FieldType.LIST,
+    buttonLabel: 'Add pattern',
+    fieldPath: topicAllowFieldPath,
+    rules: null,
+    section: 'Topics',
+    setValueOnRecipeOverride: (recipe: any, values: string[]) =>
+        setListValuesOnRecipe(recipe, values, topicAllowFieldPath),
+};
+
+const topicDenyFieldPath = 'source.config.topic_patterns.deny';
+export const TOPIC_DENY: RecipeField = {
+    name: 'topic_patterns.deny',
```

Comment:
these are going to be specific to Kafka, so you can put them in `kafka.ts` file you created.

Reply:
Done!